### PR TITLE
[DCOS-60858] [Spark Operator] Test mounting secrets

### PR DESCRIPTION
**Pull request by farhan5900**
### What changes were proposed in this pull request?
Resolves [DCOS-60858](https://jira.mesosphere.com/browse/DCOS-60858)

Implements integration tests for environment based secret as well as file based secret. It creates a secret and for the case of file based test, it mounts secret as file and check mounted path in job description. For the case of env based test, it set an environment variable by referencing it to a secret key and check for the environment variable in job description.

### Why are the changes needed?
To verify File based as well as env based secret working.

### How were the changes tested?
Ran `make test` locally.

### CODE REVIEW

File: tests/security_test.go [hunk 1]
```diff
@@ -295,3 +297,96 @@ func runTestCase(tc securityTestCase) error {
...
+	}
+
+	if strings.Contains(jobDescription, fmt.Sprintf("set to the key '%s' in secret '%s'", secretKey, secretName)) {
+		log.Infof("Successfully set environment variable to the key '%s' in secret '%s'", secretKey, secretName)
+	} else {
+		t.Errorf("Unnable to set environment variable to the key '%s' in secret '%s'", secretKey, secretName)
+	}
+}
+
+func TestFileBasedSecrets(t *testing.T) {
```
Comment (VladimirKravtsov) [inline tests/security_test.go]: These two methods(`TestEnvBasedSecret` and `TestFileBasedSecrets`)  are looking very similar. maybe it makes sense to extract the logic into another one method and invoke it with different parameters depending on test case.  
WDYT, @farhan5900 ?

Comment (farhan5900) [reply tests/security_test.go]: Right these two methods are almost same, that's why we have `runSecretTest` method containing all common logic. We can extract remaining logic as well. I will try to do that.

File: tests/security_test.go [hunk 2]
```diff
@@ -295,3 +297,93 @@ func runTestCase(tc securityTestCase) error {
...
+func runSecretTest(secretName string, secretPath string, secretKey string, expectedSecret string) error {
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		return err
+	}
+
+	client, err := utils.GetK8sClientSet()
```
Comment (rpalaznik) [inline tests/security_test.go]: After calling `spark.InstallSparkOperator()`, the `spark` struct will already have a client set ready to use as `spark.K8sClients` which you can pass to `CreateSecret` function

File: tests/security_test.go [hunk 3]
```diff
@@ -295,3 +297,93 @@ func runTestCase(tc securityTestCase) error {
...
+	err = spark.WaitUntilSucceeded(job)
+	if err != nil {
+		return err
+	}
+
+	jobDescription, err := utils.Kubectl(
+		"describe",
+		"pod",
+		"--namespace="+spark.Namespace,
+		jobName+"-driver",
```
Comment (rpalaznik) [inline tests/security_test.go]: There is a private util function that does the same https://github.com/mesosphere/kudo-spark-operator/blob/ac69995410810bd83afcff3bf3b3f4c7b4b2021a/tests/utils/job.go#L80-L82  Consider making it public and using it here. It's unlikely, but if driver pod naming changes, it's better to have only one place to fix.

File: tests/security_test.go (hunk 2 above)
Comment (farhan5900) [reply tests/security_test.go]: Thanks @rpalaznik for pointing it out. Changed it.

File: tests/security_test.go (hunk 3 above)
Comment (farhan5900) [reply tests/security_test.go]: Yeah it is better way. Made the method public.

**Pull request merged**